crates/binjs_io/src/entropy/predict.rs

```diff
@@ -336,3 +346,172 @@ impl<NodeValue> PathPredict<NodeValue, SymbolInfo> where NodeValue: Eq + Hash +
     }
 }
 
+
+
+
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+enum WindowPrediction {
+    BackReference(BackReference),
+    DictionaryIndex(DictionaryIndex),
+}
+
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct WindowPredict<NodeValue, Statistics> where NodeValue: Clone + Eq + Hash {
```

Comment:
please add comment what this is, like:
  * what kind of window (what is this for)
  * what the target sequence is
  * where the "width" points
  * how prediction is performed there
  * what the relation between window and dictionary is

after reading the following code, this looks like cache with a vector that holds entries in MRU order,
and width is the max number of cached entry,
and back reference is the index from the MRU entry.
is it correct?

Reply:
Indeed.